csrc/python_frontend/python_bindings.cpp

```diff
@@ -2327,11 +2329,19 @@ void initNvFuserPythonBindings(PyObject* module) {
             shape.end(),
             output_shape_states.begin(),
             [&fd](const Scalar& s) { return fd->recordingState(s()); });
+        std::vector<State> arg_states = {
+            fd->recordingState(minval()),
+            fd->recordingState(maxval()),
+        };
+        if (rng_seed.has_value()) {
+          TORCH_CHECK(
+              rng_offset.has_value(),
+              "When providing rng_seed, rng_offset must also be provided");
+          arg_states.push_back(fd->recordingState(rng_seed.value()()));
+          arg_states.push_back(fd->recordingState(rng_offset.value()()));
+        }
```

Comment:
@kevinstephano This is an instance of a pattern I hadn't seen yet. The `rng_seed` and `rng_offset` args are optional `Scalar`s. I am handling them by pushing them onto the back of `args_` since this automatically includes them in hashing and equality checks, but this feels kludgy since there are other possibly-positional args (`dtype` in this case) that come before `rng_seed` and `rng_offset`. I was thinking it would make sense here to make these kwonly args, but there is still the issue of how to handle them on the `RecordFunctor` side.

It seems like cases like this have been thought of to some degree, due to the presence of the `RecordFunctor::arg_names_` vector, which this PR provides a setter for. Currently I think that works alright in this case. It probably is always fine as long as classes first call `RecordFunctor::print()` then print their additional args only as kwargs.

Reply:
I fully support making `rng_seed` and `rng_offset` keyword-only arguments on the user-facing side.